utils/submit_tracks.py

```diff
@@ -0,0 +1,186 @@
+
+import sys
+import json
+import requests
+
+track_api_root = 'http://submit-tracks.review.ensembl.org/api/tracks'
+
+def submit_track(track_data):
+  request = requests.post(f"{track_api_root}/track", json=track_data)
+  if request.status_code != 201:
+    error = request.content.decode()
+    if 'Track already exists' in error:
+      print(f"Track {track_data['trigger'][-1]} already exists, skipping.")
+      return
+    print(f"Error submitting track ({request.status_code}): {error}")
+    print(f"Payload: {track_data}")
+    exit(1)
+  print(request.content.decode()) #expected response: {"track_id": "some-uuid"}
+
+def parse_csv(path):
+  if not path or not path.endswith('.csv'):
+    print('Input CSV file missing')
+    exit(1)
+  with open(path) as f:
+    lines = f.readlines()
+  lines.pop(0) #remove header
+  return [line.strip().split(',') for line in lines]
+
+mode, input = (sys.argv[1], sys.argv[2]) if len(sys.argv) > 2 else (None, None)
+
+if(mode == 'variation'):
+  if not input or not input.endswith('.json'):
+    print('Input JSON file missing')
+    exit(1)
+  with open(input) as f:
+    input_data = json.load(f)
+
+  print(f"Submitting {len(input_data)} variation tracks:")  
+  for uuid in input_data:
+    print(f"Species {uuid}:")
+    for field in ['label','datafiles','description','source']:
+      if field not in input_data[uuid] or not input_data[uuid][field]:
+        print(f"Missing field in input JSON: {uuid}->{field}")
+        exit(1)
+    variation_track = {
+      **input_data[uuid],
+      "genome_id": uuid,
+      "category": {
+        "track_category_id": "short-variants",
+        "label": "Short variants by resource",
+        "type": "Variation"
+      },
+      "trigger": ["track","expand-variation"],
+      "type": "variant",
+      "display_order": 1000,
+      "on_by_default": True
+    }
+    variation_track['sources'] = [variation_track.pop('source')]
+    submit_track(variation_track)
+
+elif(mode == 'genomic'):
+  lines = parse_csv(input)
+  print(f"Submitting {len(lines)*6} genomic tracks:")
+  for fields in lines:
+    if len(fields) < 8:
+      print(f"Invalid line in input CSV ({len(fields)} fields): {fields}")
+      exit(1)
+
+    print(f"Species {fields[2]}:")
+    method = 'annotated by' if fields[4] == 'Annotated' else 'imported from'
+    gene_pc_fwd = {
+      "genome_id": fields[2],
+      "label": "Protein coding genes",
+      "category": {
+        "track_category_id": "genes-transcripts",
+        "label": "Genes & transcripts",
+        "type": "Genomic"
+      },
+      "trigger": ["track","gene-pc-fwd"],
+      "type": "gene",
+      "colour": "DARK_GREY",
+      "additional_info": "Forward strand",
+      "display_order": -110,
+      "on_by_default": True,
+      "datafiles": {},
+      "description": f"Genes {method} {fields[3]}",
+      "sources": [{ "name": fields[3], "url": fields[5] }]
+    }
+    submit_track(gene_pc_fwd)
+
+    gene_pc_rev = {
+      **gene_pc_fwd,
```

Comment:
Bad:

![image](https://github.com/Ensembl/ensembl-web-track-api/assets/6834224/abd6e0a7-770a-4354-8406-5d067192aa3f)

see beta for good 🙂 Also, notice that reference sequence and %gc don't have those

Reply:
Track data reloaded.